redisinsight/ui/src/utils/formatters/java-date.ts

```diff
@@ -14,6 +14,23 @@ export default class JavaDate implements JavaSerializable {
   }
 
   readResolve() {
-    return new Date(Number(this.time))
+    let timeValue: number
+
+    // Handle two's complement conversion for negative numbers
+    if (this.time > 9223372036854775807n) {
+      // If the number is larger than MAX_LONG, it's a negative number in two's complement
+      timeValue = Number(this.time - 18446744073709551616n)
```

Comment:
@GnaneshKunal could you please add 9223372036854775807n and 18446744073709551616n as constant variables?

Reply:
Done.